back/engines/commercial/insights/spec/services/insights/category_detection_service_spec.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Insights::CategoryDetectionService do
+  describe '#detect_categories' do
+    subject(:service) { described_class.new(nlp_client) }
+    let(:nlp_client) { instance_double(NLP::Api, 'nlp_client') }
+
+    let(:response) {[
+      {
+        "text" => "kaggle",
+        "frequency" => 11
+      },
+      {
+        "text" => "google",
+        "frequency" => 10
+      }
+    ]}
+
+    context 'no input' do
+      let(:view) { create(:view) }
+      it 'returns [] when there\'s no inputs' do
+        allow(nlp_client).to receive(:project_tag_suggestions).and_return(response)
+        result = service.detect_categories(view)
+
+        expect(result.length).to eq(0)
```

Comment:
```suggestion
        expect(result).to eq []
```
as in the `it` description?